src/rgw/rgw_sal.cc

```diff
@@ -122,6 +126,14 @@ rgw::sal::Store* StoreManager::init_storage_provider(const DoutPrefixProvider* d
       delete store;
       return nullptr;
     }
+    if (store->initialize(cct, dpp) < 0) {
```

Comment:
Here we will always get 0 (because there is only return 0 at least for rgwradosstore), Is this condition necessary?

Reply:
seems like a wise condition to me...

Reply:
Yeah, I'd like the boilerplate to be here, so that if we change RadosStore::initialize, it just works.  Also, at some point, we're likely to combine the code for various stores.